### ISSUE TITLE
Result return value in doctest is not supported

### ISSUE THREAD

**Issue by upsuper** [intellij-rust/intellij-rust#5097]
## Environment

* **IntelliJ Rust plugin version:** 0.2.117.2170-193
* **Rust toolchain version:** 1.42.0
* **IDE name and version:** CLion 2019.3.4
* **Operating system:** Ubuntu 19.10

## Problem description
Tailing result value in doctest like `Ok(())` is incorrectly marked as mismatch types while Rust compiler accepts it.

## Steps to reproduce

Put the following doctest somewhere:
```
/// ```
/// Ok::<(), Box<dyn std::error::Error>>(())
/// ```
```

The IDE reports "mismatched types [E308]", "expected `()`, found `Result<(), <unknown>>`, but Rust compiler is fine with it.

This ability was added in 1.34.0, see [Using `?` in doc tests](https://doc.rust-lang.org/rustdoc/documentation-tests.html#using--in-doc-tests) in the rustdoc document.

**Comment by lovasoa** [comment]
I also need that for [good_lp](https://github.com/lovasoa/good_lp). Any pointer on how to fix this ?

**Comment by xamgore** [comment]
The rustdoc book:

> As of version 1.34.0, one can also omit the `fn main()`, but you will have to disambiguate the error type:
> 
> ```rust
> /// ```rust
> /// use std::io;
> /// let mut input = String::new();
> /// io::stdin().read_line(&mut input)?;
> /// # Ok::<(), io::Error>(())
> /// ```
> ```
> 
> This is an unfortunate consequence of the `?` operator adding an implicit conversion, so type inference fails because the type is not unique. Please note that you must write the `(())` in one sequence without intermediate whitespace so that `rustdoc` understands you want an implicit `Result`-returning function.

So, let's try it:

<img width="508" alt="image" src="https://github.com/user-attachments/assets/f3c5c630-d2de-407c-ab00-63cc54bd1cc3">

Tests compile, though. Absolutely a bummer for open source development.
